process/coordinator/process.go

```diff
@@ -667,6 +677,19 @@ func (tc *transactionCoordinator) CreateMbsAndProcessCrossShardTransactionsDstMe
 	return miniBlocks, nrTxAdded, allMBsProcessed, nil
 }
 
+func (tc *transactionCoordinator) revertIfNeeded(txsToBeReverted [][]byte) {
+	shouldRevert := tc.shardCoordinator.SelfId() == core.MetachainShardId && len(txsToBeReverted) > 0
```

Comment:
so no revert at all for metachain ? for miniblocks processing ?

Reply:
They are reverted, as this method is called by metachain with all txs already processed from successful mbs execution when complete processing of the whole shard hdr fails.